src/registrar/management/commands/load_transition_domain.py

```diff
@@ -0,0 +1,463 @@
+
+    def print_debug_mode_statements(self, debug_on, debug_max_entries_to_parse):
+        if debug_on:
+            logger.info(
+                f"""{termColors.OKCYAN}
+            ----------DEBUG MODE ON----------
+            Detailed print statements activated.
+            {termColors.ENDC}
+            """
+            )
+        if debug_max_entries_to_parse > 0:
+            logger.info(
+                f"""{termColors.OKCYAN}
+            ----------LIMITER ON----------
+            Parsing of entries will be limited to
+            {debug_max_entries_to_parse} lines per file.")
+            Detailed print statements activated.
+            {termColors.ENDC}
+            """
+            )
+
+    def get_domain_user_dict(self, domain_statuses_filename, sep):
+        """Creates a mapping of domain name -> status"""
+        # TODO: figure out latest status
+        domain_status_dictionary = defaultdict(str)
+        # NOTE: how to determine "most recent" status?
+        logger.info("Reading domain statuses data file %s", domain_statuses_filename)
+        with open(domain_statuses_filename, "r") as domain_statuses_file:  # noqa
+            for row in csv.reader(domain_statuses_file, delimiter=sep):
+                domainName = row[0].lower()
+                domainStatus = row[1].lower()
+                # print("adding "+domainName+", "+domainStatus)
+                domain_status_dictionary[domainName] = domainStatus
+        logger.info("Loaded statuses for %d domains", len(domain_status_dictionary))
+        return domain_status_dictionary
+
+    def get_user_emails_dict(self, contacts_filename, sep):
+        """Creates mapping of userId -> emails"""
+        # NOTE: is this one to many??
+        user_emails_dictionary = defaultdict(list)
+        logger.info("Reading domain-contacts data file %s", contacts_filename)
+        with open(contacts_filename, "r") as contacts_file:
+            for row in csv.reader(contacts_file, delimiter=sep):
+                userId = row[0]
+                user_email = row[6]
+                user_emails_dictionary[userId].append(user_email)
+        logger.info("Loaded emails for %d users", len(user_emails_dictionary))
+        return user_emails_dictionary
+
+    def get_mapped_status(self, status_to_map):
+        # Map statuses as follows;
+        # "serverHold” fields will map to hold clientHold to hold
+        # and any ok state should map to Ready.
+        # Check if there are any statuses that are not
+        # serverhold, client hold or OK in the original data set.
+        status_maps = {
+            "hold": TransitionDomain.StatusChoices.HOLD,
+            "serverhold": TransitionDomain.StatusChoices.HOLD,
+            "clienthold": TransitionDomain.StatusChoices.HOLD,
+            "created": TransitionDomain.StatusChoices.READY,
+            "ok": TransitionDomain.StatusChoices.READY,
+        }
+        return status_maps[status_to_map]
```

Comment:
**question (non-blocking)**: See above. Do we want to use `status_maps.get(status_to_map)` to avoid `KeyError` here and propagate `None` for statuses not in `status_maps`?

Reply:
Yes to avoiding keyError!

Reply:
Good point.  I will update